src/extension/rpc.ts

```diff
@@ -0,0 +1,191 @@
+import { NoInfer, SafeAny } from "../types";
+import { MessageAdapter } from "./messageAdapters";
+import {
+  ApolloClientDevtoolsRPCMessage,
+  MessageType,
+  isRPCMessage,
+} from "./messages";
+
+type RPCParams = Record<string, unknown>;
+
+type RPCRequestMessageFormat = {
+  type: string;
+  params: RPCParams;
+};
+
+type RPCResponseMessageFormat =
+  | { sourceId: number; result: unknown }
+  | { sourceId: number; error: unknown };
+
+export type RPCMessage<
```

Comment:
I don't know if there is a better way to do this. This is really just a container type to hold the values so that we can unwrap them in the various usages below (`RpcClient` and `createRPCHandler`)